tpcx_bb/queries/q17/tpcx_bb_query_17_sql.py

```diff
@@ -0,0 +1,125 @@
+#
+# Copyright (c) 2019-2020, NVIDIA CORPORATION.
+# Copyright (c) 2019-2020, BlazingSQL, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+
+
+from blazingsql import BlazingContext
+from xbb_tools.cluster_startup import attach_to_cluster
+from dask_cuda import LocalCUDACluster
+from dask.distributed import Client
+import os
+
+from xbb_tools.utils import (
+    benchmark,
+    tpcxbb_argparser,
+    run_bsql_query,
+)
+
+cli_args = tpcxbb_argparser()
+
+# ------- Q17 ------
+q17_year = 2001
+q17_month = 12
+q17_i_category_IN = "('Books', 'Music')"
+date_string = str(q17_year) + "-" + str(q17_month) + "-"
+
+@benchmark(
+    compute_result=cli_args["get_read_time"], dask_profile=cli_args["dask_profile"]
+)
+def read_tables(data_dir):
+    bc.create_table("store_sales", data_dir + "store_sales/*.parquet")
+    bc.create_table("item", data_dir + "item/*.parquet")
+    bc.create_table("customer", data_dir + "customer/*.parquet")
+    bc.create_table("store", data_dir + "store/*.parquet")
+    bc.create_table("date_dim", data_dir + "date_dim/*.parquet")
+    bc.create_table("customer_address", data_dir + "customer_address/*.parquet")
+    bc.create_table("promotion", data_dir + "promotion/*.parquet")
+
+@benchmark(dask_profile=cli_args["dask_profile"])
+def main(data_dir, client):
+    read_tables(data_dir)
+
+    dates_pandas = (
+        bc.sql(
+            """select min(d_date_sk) as start_date, max(d_date_sk) as end_date
+    from date_dim
+    where d_year = """
+            + str(q17_year)
+            + """ and d_moy = """
+            + str(q17_month)
+            + """
+    """
+        )
+        .compute()
+        .to_pandas()
+    )
+    date_start_sk = dates_pandas["start_date"][0]
+    date_end_sk = dates_pandas["end_date"][0]
+
+    query = (
+        """
```

Comment:
Could we also use format strings here, too?

Reply:
Sure.